### PR TITLE
Integrate MRU tracker with element creation workflow

### PR DESCRIPTION
**Pull request by Amadeus1791**
This Pr connects the `MostRecentlyUsedTracker` (from PR #1799 ) to the element creation workflow. Now when users create function blocks via the search dialog, the tracker automatically records usage for later retrieval.

Changes:
- Add MRU tracker instance to `NewInstanceCellEditor`
- Initialize tracker in `setTypeLibrary()` with project from TypeLibrary
- Record usage in `doGetValue()` when element is selected
- Add debug output to verify tracking is working

This enables automatic MRU tracking without requiring any user action. Debug output will now appear in the console showing MRU list updates as users create function blocks.

Depends on: PR #1 (MRU infrastructure)

### How to Verify

1. Open an FB editor in a project.
2. Create FBs via type-to-search
3. Check console for debug prints
4. Verify per-project prefs file (project/.settings/org.eclipse.fordiac.ide.gef.prefs) updates with the MRU list.
5. Test edge cases: Delete project -> no errors; new project -> empty MRU.

UI display of MRU items will come in follow-up PR

### CODE REVIEW

**Comment by azoitl** [comment]
@Amadeus1791 can you please rebase your commit on top of develop and force push it so that we get a cleaner history. This would help me to review and test.

File: plugins/org.eclipse.fordiac.ide.gef/src/org/eclipse/fordiac/ide/gef/editors/NewInstanceCellEditor.java [hunk 1]
```diff
@@ -85,6 +88,19 @@ public Button getMenuButton() {
 
 	public void setTypeLibrary(final TypeLibrary typeLib) {
 		paletteFilter = new PaletteFilter(typeLib);
+
+		// Initialize MRU tracker for the current project
+		if (typeLib != null && mruTracker == null) {
+			try {
+				final IProject project = typeLib.getProject();
+				if (project != null && project.exists()) {
+					mruTracker = new MostRecentlyUsedTracker(project);
```
Comment (azoitl) [inline plugins/org.eclipse.fordiac.ide.gef/src/org/eclipse/fordiac/ide/gef/editors/NewInstanceCellEditor.java:97]: I think the MRU Tracker should be a field of the TypeLibrary. Ohterwise any Cell editor would track its own most recently used items

**Review by azoitl — CHANGES_REQUESTED** [review]

**Comment by Amadeus1791** [comment]
Closing this PR. Usage recording is now consolidated into the element-creation path, and
the tracker is owned by the TypeLibrary (as suggested by @azoitl) both in #2769. Thanks!
